File: benchmarks/baselines/cross-language/sieve.js
const n = 1_000_000;
const is_prime = new Uint8Array(n + 1).fill(1);
is_prime[0] = is_prime[1] = 0;
for (let p = 2; p * p <= n; p++) {
    if (is_prime[p]) {
        for (let m = p * p; m <= n; m += p) is_prime[m] = 0;
    }
}
let count = 0;
for (let i = 0; i <= n; i++) if (is_prime[i]) count++;
process.exit(count === 78498 ? 0 : 1);

File: benchmarks/baselines/cross-language/sieve.rs
fn sieve(limit: usize) -> Vec<bool> {
    let mut is_prime = vec![true; limit + 1];
    is_prime[0] = false;
    if limit >= 1 { is_prime[1] = false; }
    let mut p = 2;
    while p * p <= limit {
        if is_prime[p] {
            let mut m = p * p;
            while m <= limit {
                is_prime[m] = false;
                m += p;
            }
        }
        p += 1;
    }
    is_prime
}

fn main() {
    let n = 1_000_000;
    let primes = sieve(n);
    let count = primes.iter().filter(|&&b| b).count();
    std::process::exit(if count == 78498 { 0 } else { 1 });
}
